File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_map<int,bool> seen;
        for(int i=0;i<arr.size();++i){
            if (seen.count(2*arr[i]) || (arr[i]%2 ==0 && seen.count(arr[i]/2))){
                return true;
            }
            else{
                seen[arr[i]]=true;
            }
        }
        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/1468-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        seen = {}
        for a in arr:
            if 2*a in seen or (a%2==0 and a/2 in seen):
                return True
            else:
                seen[a] = True
        return False
                